### PR TITLE
Style webview to display baked content

### PR DESCRIPTION
**Pull request by helenemccarron**
# TODO
## Main content style changes
- [x] Style the  headers within body  (decrease size, color, unbold, vertical spacing) (may already be fixed) (hide the black title on each cooked page, keep the blue one `h2:first`)
- [x] Figure captions/numbering: styled, aligned with figure (the width of the caption should not exceed the width of the image)
- [x] width, color for figure 1.2 + title + caption
- [x] Summary: remove toggle, the summary is now handled in the content. We might need styling (maybe remove it if there is no summary; see https://github.com/Connexions/cnx-epub/pull/95 and https://github.com/Connexions/cnx-recipes/pull/83 and [example](http://cte-cnx-dev.cnx.org/contents/MBiUQmmY@19.2:kcV4GRqc@6/Preface) ) [abstract copied into the content](http://cte-cnx-dev.cnx.org/contents/aWGdK2jw@11.330:6RH0nLs4@8/11--What-Is-Economics-and-Why-)
- [ ] Fix spacing for Try-Its and Examples in Intro to Stats [example](http://cte-cnx-dev.cnx.org/contents/MBiUQmmY@19.2:CY4aJuYS@13/12--Data-Sampling-and-Variatio)
- [ ] Style "By the end of this section, you will be able to:" (maybe?)
- [ ] Style all features to match uncooked feature styling.
## Design for end of chapter, end of book
- [x] Key concepts: styling + spacing for section titles (vertical spacing [example](http://cte-cnx-dev.cnx.org/contents/aWGdK2jw@11.330:S5CE05DP@1/Key-Concepts-and-Summary) )
  - [x] References: have the chapter number with title, (a page per chapter, if possible)
    - [example](http://cte-cnx-dev.cnx.org/contents/aWGdK2jw@11.330:QsW_NPNh@1/References) just change vertical spacing on the Header
- [x] Exercises: restyled, numbering: 1.  + spacing [links to ] [example](http://cte-cnx-dev.cnx.org/contents/aWGdK2jw@11.330:eVUNeZSE@1/Review-Questions)
  - TODO: Open an Issue to change exercise divs to be some other kind of element. Unsure what the element should be
- [x] Styling ordered list > hide double numbering, this includes footnotes https://github.com/Connexions/rhaptos.cnxmlutils/pull/150 and https://github.com/Connexions/webview/pull/1464 and https://github.com/Connexions/rhaptos.cnxmlutils/pull/151
- [x] Index: style letter, padding between links, coma between them, divider between term and links, bold + padding between entries (https://github.com/Connexions/cnx-recipes/pull/79)
  - vertical spacing between blocks, make the letter bigger.
## Editorial decisions
- [x] End of chapter sections in the navigation shouldn't be numbered.

### CODE REVIEW

File: src/scripts/modules/media/body/content.less [hunk 1]
```diff
@@ -68,18 +70,29 @@ section > figure {
...
 }
 
-figure {
+.figure {
+  display: table;
+  .caption-container {
+    padding-top: 1rem;
+    display: table-caption;
+    caption-side: bottom;
+    .figure-number {
```
Comment (RoyEJohnson) [inline src/scripts/modules/media/body/content.less]: Formatting problems here

**Comment by philschatz** [comment]
I believe the following are done, right @helenemccarron @zroehr ?

The selectors to style the exercise solutions were hacky & inconsistent with how to number exercise problems (probably would require a change to cnx-recipes)

# Screenshots

[Key concepts: styling + spacing for section titles](http://localhost:8000/contents/aWGdK2jw@11.330:3ej9ziSN@1/Key-Concepts-and-Summary)
image


[Exercises: restyled, numbering:](http://localhost:8000/contents/aWGdK2jw@11.330:W73klFIP@1/Self-Check-Questions)
image

... and solutions:

image

**Comment by helenemccarron** [comment]
It looks like the page titles are still repeating, but maybe it will be fixed with @zroehr's typography push? 

What are the inconsistencies for the exercises and what solutions  do you think would be useful?

**Comment by philschatz** [comment]
Currently, the selectors are `[data-type="problem"] > a.number` and `.solutions-container [data-type="solution"] > .ui-body > a /*> .number*/` . The inconsistencies in the selectors that I see are:

- the Problem contains an `a.number` but the Solution contains an `a > .number`
- the `[data-type="solution"]` contains a `ui-toggle-wrapper` (maybe it's from webview JS?)
- `.ui-body` is only in the solution (maybe it's from webview JS?)

**Comment by helenemccarron** [comment]
I can look into the first one, I thought they were consistent but it's possible something like that slipped through. 

For the next two points, yes, those are generated by webview and not the ruleset

**Comment by philschatz** [comment]
I just pushed a fix for items 2&3 (from previous comment) and https://github.com/Connexions/cnx-recipes/pull/85 should address the 1st item

**Comment by philschatz** [comment]
[CSS changes to webview](https://trello.com/c/cJbbOtqS/371-css-changes-to-webview)

File: src/scripts/modules/media/body/content-cooked.less [hunk 2]
```diff
@@ -167,3 +168,71 @@ h1.example-title .text {
...
+  h2.os-title {
+    font-size: 2.1rem;
+    font-weight: bold;
+  }
+  .os-number {
+    padding-right: 0;
+    font-weight: bold;
+    text-decoration: none;
+    &::after {
+      content: ".";
```
Comment (philschatz) [inline src/scripts/modules/media/body/content-cooked.less]: Should this period be added in the recipes? /cc @zroehr @helenemccarron

Comment (helenemccarron) [inline src/scripts/modules/media/body/content-cooked.less]: Yes, I need to add it to the rulesets.

**Pull request merged**